string   target_type
uint32   tag_rows
uint32   tag_cols
float64  tag_size
float64  tag_spacing
